[lldb/docs] Breakdown python reference into multiple files

medismailben opened the pull request:
This pages improve the LLDB website documentation readability and discoverability by breaking down the very long python-reference page into multiple subpages each explaining a specific topic.

The long term goal is to have tutorials for every scripting extension.

This also converts the pages to markdown, since it's easier to write.

JDevlieghere reviewed (approved):
LGTM. There's tons of opportunity for improvement but that shouldn't block this PR.

medismailben commented:
> Any reason we can't use the existing [.htaccess](https://github.com/llvm/llvm-project/blob/main/lldb/docs/.htaccess) for that?

Sure, we can also do that

kastiglione commented on lldb/docs/use/tutorials/writing-custom-commands.md:
```diff
@@ -0,0 +1,431 @@
+import subprocess
+
+def ls(debugger, command, result, internal_dict):
+    proc = subprocess.run(["/bin/ls"] + command.split(), capture_output=True, text=True)
+    result.PutCString(proc.stdout)
```
I always print to the `result` as a file, and have assumed that to be the idiomatic way to print to the `result` in python.

```suggestion
    output = subprocess.check_output(["/bin/ls"] + command.split(), text=True)
    print(output, file=result)
```

kastiglione reviewed (approved):
one more comment about one of the updated examples

medismailben committed:
[lldb/docs] Breakdown python reference into multiple files

This pages improve the LLDB website documentation readability and
discoverability by breaking down the very long python-reference page
into multiple subpages each explaining a specific topic.

The long term goal is to have tutorials for every scripting extension.

This also converts the pages to markdown, since it's easier to write.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>
